src/graph/test/reify/calculate-save-value.ts

```diff
@@ -0,0 +1,73 @@
+import { Spec } from '@vltpkg/spec'
+import t from 'tap'
+import { calculateSaveValue } from '../../src/reify/calculate-save-value.ts'
+
+t.test('non-registry type, just saved as-is', t => {
+  const s = Spec.parse('x', 'github:a/b')
+  t.equal(calculateSaveValue('git', s, 'a/b', '1.2.3'), 'github:a/b')
+  t.end()
+})
+
+// registry cases
+// nodeVersion is always 1.2.3
+const cases: [
+  spec: string,
+  existing: string | undefined,
+  expect: string,
+  comment: string,
+][] = [
+  ['1', undefined, '1', 'new dep, use specified range'],
+  ['', undefined, '^1.2.3', 'new dep, no spec, use caret range'],
```

Comment:
maybe add a few more variety of ranges, even though it should be the same code path I think it might guard against future regressions, I mean things like `>2.1 || 3`, `~1.1.0`, `1.x.x`